### PR TITLE
[Enhancement] Enhance superset's compatibility with external catalog

### LINKED ISSUE #16313: Let Superset query the external table through catalog.table
support 
```
select database();
```
should return catalogName.databaseName in external catalog
```
show schemas;
```
should return catalogName.databaseName in external catalog
```
select xxx from `catalogName.databaseName`.`table` 
```
should be supported

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16313

## Problem Summary(Required) ：

For external table
```
mysql> select database();
+--------------------+
| DATABASE()         |
+--------------------+
| glue_test.tmp_test |
+--------------------+
1 row in set (0.13 sec)
```
```
mysql> show schemas;
+----------------------------------+
| Database                         |
+----------------------------------+
| glue_test.default                |
| glue_test.delta_db               |
| glue_test.hudi_db                |
| glue_test.ice_db                 |
| glue_test.iceberg_db             |
| glue_test.tmp_test               |
+----------------------------------+
```
```
mysql> show full tables from glue_test.tmp_test;
+----------------------------+------------+
| Tables_in_tmp_test         | Table_type |
+----------------------------+------------+
| test_map                   | BASE TABLE |
| test_map2                  | BASE TABLE |
| test_map3                  | BASE TABLE |
| test_struct                | BASE TABLE |
+----------------------------+------------+
```
```
select * from `glue_test.tmp_test`.test_table 
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 18 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|6|33.33%|[5449, 5450, 5451, 5452]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|4|8|50.00%|[568, 615, 616, 617]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|4|75.00%|[1136]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -5442,6 +5443,14 @@ private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
         if (parts.size() == 3) {
             return new TableName(parts.get(0), parts.get(1), parts.get(2));
         } else if (parts.size() == 2) {
+            String pendingDbName = qualifiedName.getParts().get(0);
+            // support query `external_catalog.database`
+            if (pendingDbName.contains(".")) {
+                List<String> catalogAndDb = Splitter.on('.').trimResults().splitToList(pendingDbName);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:5449]: Validate the list size?

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 2]
```diff
@@ -601,6 +609,13 @@ private void handleShowTable() throws AnalysisException {
             catalogName = ctx.getCurrentCatalog();
         }
         String dbName = showTableStmt.getDb();
+
+        // support show full tables from `external_catalog.database`
+        if (dbName.contains(".")) {
+            List<String> catalogAndDb = Splitter.on('.').trimResults().splitToList(dbName);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:615]: Validate size of list?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 21 (52.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|7|28.57%|[5449, 5450, 5451, 5452, 5453]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|4|9|44.44%|[568, 615, 616, 617, 618]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|5|5|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Pull request closed without merging**